irods/test/collection_test.py

```diff
@@ -322,6 +325,52 @@ def test_register_collection_with_checksums(self):
         shutil.rmtree(dir_path)
 
 
+    def test_collection_with_trailing_slash__323(self):
+        Home = helpers.home_collection(self.sess)
+        subcoll, dataobj = [unique_name(my_function_name(),time.time()) for x in range(2)]
+        subcoll_fullpath = "{}/{}".format(Home,subcoll)
+        subcoll_unnormalized = subcoll_fullpath + "/"
+        try:
+            # Test create and exists with trailing slashes.
+            self.sess.collections.create(subcoll_unnormalized)
+            c1 = self.sess.collections.get(subcoll_unnormalized)
+            c2 = self.sess.collections.get(subcoll_fullpath)
+            self.assertEqual(c1.id, c2.id)
+            self.assertTrue(self.sess.collections.exists(subcoll_unnormalized))
+
+            # Test data put to unnormalized collection name.
+            with open(dataobj, "wb") as f: f.write(b'hello')
+            self.sess.data_objects.put(dataobj, subcoll_unnormalized)
+            self.assertEqual(
+                self.sess.query(DataObject).filter(DataObject.name == dataobj).one()[DataObject.collection_id]
+               ,c1.id
+            )
+        finally:
+            if self.sess.collections.exists(subcoll_fullpath):
+                self.sess.collections.remove(subcoll_fullpath, recurse = True, force = True)
+            if os.path.exists(dataobj):
+                os.unlink(dataobj)
+
+
+    def test_object_paths_with_dot_and_dotdot__323(self):
+
+        normalize = iRODSCollection.normalize_path
+        session = self.sess
+        home = helpers.home_collection( session )
+
+        # Test requirement for collection names to be absolute
+        with self.assertRaises(iRODSCollection.AbsolutePathRequired):
+            normalize('../public')
+
+        # Test '.' and double slashes
+        public_home = normalize(home,'..//public/.//')
+        self.assertEqual(public_home, '/{sess.zone}/home/public'.format(sess = session))
+
+        # Test that '..' cancels last nontrival path element
+        subpath = normalize(home,'./collA/coll2/./../collB')
+        self.assertEqual(subpath, home + "/collA/collB")
+
```

Comment:
please add two more cases where there are multiple '..' somewhere in the path...
 - both right next to one another 
 - spread out by a couple elements

Reply:
done.